scripts/tools/IndexMissingRecords.php

```diff
@@ -0,0 +1,205 @@
+            }
+
+            if (!$this->isIndexed($advancedSearch, $this->getIndexName($class), $resource->getUri())) {
+                $missingResources[$resource->getUri()] = $resource;
+
+                $missingIndexReport->add(
+                    Report::createInfo(
+                        $resource->getUri() . ' (' . $resource->getLabel() . ')'
+                    )
+                );
+            }
+        }
+
+        if ($reindex) {
+            $reIndexedReport = Report::createSuccess('ReIndexed resources');
+            $mainReport->add($reIndexedReport);
+
+            $indexer = $this->getSyncResourceResultIndexer();
+
+            foreach ($missingResources as $resource) {
+                $reIndexedReport->add(
+                    Report::createInfo(
+                        $resource->getUri() . ' (' . $resource->getLabel() . ')'
+                    )
+                );
+
+                $indexer->addIndex($resource);
+
+                $missingResourcesIndexed[$resource->getUri()] = $resource->getLabel();
+            }
+        }
+
+        $summaryReport = Report::createSuccess('Summary');
+        $summaryReport->add(Report::createInfo('Missing resources: ' . count($missingResources)));
+        $summaryReport->add(Report::createInfo('Missing resources indexed: ' . count($missingResourcesIndexed)));
+
+        $mainReport->add($summaryReport);
+
+        return $mainReport;
+    }
+
+    private function isIndexed(ElasticSearch $search, string $index, string $uri)
+    {
+        $query = new Query($index);
+        $query->addCondition('_id:"' . $uri . '"');
+        $query->setLimit(1);
+
+        return $search->search($query)->getTotalCount() > 0;
+    }
+
+    private function getIndexName(string $classUri): ?string
+    {
+        //@TODO Remove direct call for ElasticSearch
+        return IndexerInterface::AVAILABLE_INDEXES[$classUri] ?? null;
+    }
+
+    private function search(string $classUri, int $offset, int $limit): ResultSetInterface
+    {
+        $search = $this->getComplexSearchService();
+
+        $queryBuilder = $search->query();
+
+        $criteria = $search->searchType($queryBuilder, $classUri, true);
+
+        $queryBuilder = $queryBuilder->setCriteria($criteria);
+
+        $queryBuilder->setOffset($offset);
+
+        if ($limit > 0) {
+            $queryBuilder->setLimit($limit);
+        }
+
+        return $search->getGateway()->search($queryBuilder);
+    }
+
+    private function getElasticSearch(): ElasticSearch
+    {
+        return $this->getServiceLocator()->get(SearchProxy::SERVICE_ID)->getAdvancedSearch();
+    }
+
+    private function getComplexSearchService(): ComplexSearchService
+    {
+        return $this->getServiceLocator()->get(ComplexSearchService::SERVICE_ID);
+    }
+
+    private function getSyncResourceResultIndexer(): SyncResourceResultIndexer
+    {
+        return $this->getServiceManager()->get(SyncResourceResultIndexer::class);
+    }
```

Comment:
Why you are not using new DI container? 🙂

Reply:
Good point :). Cause we might backport this to the next UDIR deployment, so I wanted to avoid more conflicts.